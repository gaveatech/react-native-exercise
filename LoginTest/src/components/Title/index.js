import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

const Title = ({children, color}) => {
  return <Text {...{color}}>{children}</Text>;
};

const Text = styled.Text`
  font-family: 'RobotoBold';
  font-size: 24px;
  margin-bottom: 5px;
  color: ${props => props.color};
`;

Title.propTypes = {
  color: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Title.defaultProps = {
  color: '#000',
};

export default Title;
