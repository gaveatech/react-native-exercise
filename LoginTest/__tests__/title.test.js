import React from 'react';
import renderer from 'react-test-renderer';
import Title from '../src/components/Title';

describe('<Title>', () => {
  it('Snapshot', () => {
    const tree = renderer.create(<Title>Teste</Title>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
