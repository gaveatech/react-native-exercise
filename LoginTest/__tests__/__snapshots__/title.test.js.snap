// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title> Snapshot 1`] = `
<Text
  color="#000"
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "RobotoBold",
        "fontSize": 24,
        "marginBottom": 5,
      },
    ]
  }
>
  Teste
</Text>
`;
